alibi_detect/saving/schemas.py

```diff
@@ -52,19 +49,22 @@ def coerce_int2list(value: int) -> List[int]:
         return value
 
 
-def validate_model(model: Callable, values: dict) -> Callable:
-    """Validator to check the model is compatible with the given backend"""
-    backend = values['backend']
-    if backend == 'tensorflow' and not isinstance(model, SupportedModels_tf):
-        raise ValueError('A TensorFlow backend is not available for this model')
-    elif backend == 'pytorch' and not isinstance(model, SupportedModels_torch):
-        raise ValueError('A PyTorch backend is not available for this model')
```

Comment:
I realised that these `ValueError`'s did not necessarily explain what we are actually validating. The validation was intended to check that the given `model` matched `backend`, but the error message suggested something else. We can reword if we decide to still validate the `model` type against `backend` (see next comment).